### PR TITLE
Store the result in the return local in native-to-managed IL stubs.

### LINKED ISSUE #65177: [mono] Tests failing with error: Invalid IL code in (wrapper native-to-managed) System.Runtime.InteropServices.PosixSignalRegistration:HandlerRoutine (int): IL_0049: ret
Basically all the libraries tests on mono on Windows seem to be failing in recent runtime-extra-platform builds with:
```
error: Invalid IL code in (wrapper native-to-managed) System.Runtime.InteropServices.PosixSignalRegistration:HandlerRoutine (int): IL_0049: ret       
```

Build: https://dev.azure.com/dnceng/public/_build/results?buildId=1605729&view=results
Example log: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-9842560bc0804bd281/System.Linq.Expressions.Tests/1/console.a86b0b70.log?sv=2019-07-07&se=2022-03-02T21%3A53%3A33Z&sr=c&sp=rl&sig=l0AXWan5SR487aw3xXpV1Px49%2FlM36Kt0lQym%2FPf3uo%3D

|Build|Kind|Start Time|
|---|---|---|
[1605729](https://dev.azure.com/dnceng/public/_build/results?buildId=1605729)|Rolling|2022-10-02|
[1607081](https://dev.azure.com/dnceng/public/_build/results?buildId=1607081)|Rolling|2022-11-02|

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #65177

Sorted includes is from the auto-formatter.

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
> Sorted includes is from the auto-formatter.

I would turn this off then. Reordering includes is fraught and can destabilize things.

**Comment by jkoritzinsky** [comment]
Mono should change their formatter configuration if they don't want the includes sorted.

File: src/mono/mono/metadata/marshal-ilgen.c [hunk 1]
```diff
@@ -8,45 +8,45 @@
...
 #include "mono/utils/mono-memory-model.h"
-#include "mono/utils/atomic.h"
-#include <mono/utils/mono-threads.h>
-#include <mono/utils/mono-threads-coop.h>
+#include "mono/utils/mono-tls.h"
+#include "object.h"
+#include <errno.h>
 #include <mono/utils/mono-error-internals.h>
+#include <mono/utils/mono-threads-coop.h>
+#include <mono/utils/mono-threads.h>
```
Comment (lambdageek) [inline src/mono/mono/metadata/marshal-ilgen.c:48]: nit: slightly prefer if we could drop unrelated reformatting.

**Review by lambdageek — APPROVED** [review]

**Comment by lambdageek** [comment]
> Mono should change their formatter configuration if they don't want the includes sorted. Otherwise there's no way to enforce formatting.

How do I do that?  I'm an .editorconfig noob

**Comment by jkoritzinsky** [comment]
Removed the formatting changes for now.

I think that we'd need to switch Mono to using .clang-format instead of .editorconfig to get the options needed.

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by elinor-fung — APPROVED** [review]

**Pull request merged**